ddsmt.py

```diff
@@ -262,48 +262,44 @@ def _substitute (subst_fun, substlist, superset, randomized,
     assert (g_smtformula)
     assert (substlist in (g_smtformula.subst_scopes, g_smtformula.subst_cmds,
                           g_smtformula.subst_nodes))
+    min_gran = 0.1 * sqrt(len(superset))
```

Comment:
I think that using a minimum granularity here is in general a good idea. However, I would introduce an option for this. Let's say, by default it is the minimum granularity above, but optionally you allow tests where even for large subsets every single item is tested in the worst case. Else, it might happen that a test case where this is the only option to simplify, can not be simplified.